src/FieldNumber.tsx

```diff
@@ -0,0 +1,237 @@
+ * pebble based number field component
+ *
+ * @see https://pebble.istreamplanet.net/#/Components/FieldNumber
+ */
+const FieldNumberAdapter = ({
+  autoFocus,
+  className,
+  clearBtnFunc,
+  disabled,
+  helpText,
+  hideLabel,
+  id,
+  isReadOnly,
+  label,
+  name,
+  onFocus,
+  placeholder,
+  prefix,
+  required,
+  size,
+  suffix,
+  type,
+  width,
+}: FieldNumberAdapterOptions): JSX.Element => (
+  <Field name={name} type={type}>
+    {generateFieldNumberRenderProp({
+      autoFocus,
+      className,
+      clearBtnFunc,
+      disabled,
+      helpText,
+      hideLabel,
+      id,
+      isReadOnly,
+      label,
+      onFocus,
+      placeholder,
+      prefix,
+      required,
+      size,
+      suffix,
+      width,
+    })}
+  </Field>
+);
+
+interface FieldNumberAdapterOptions extends InputOptions {
+  /**
+   * Automatically focuses the input element when true
+   */
+  autoFocus?: boolean;
+  /**
+   * Classname applied/passed to the FieldText pebble component
+   */
+  className?: string;
+  /**
+   * Callback for a button to clear the input value
+   */
+  clearBtnFunc?: Function;
+  /**
+   * Flag that disables the input and makes it not focusable when true
+   */
+  disabled?: boolean;
+  /**
+   * Map of the existing errors in the Formik form state
+   */
+  errors?: object;
+  /**
+   * Boolean flag that will hide the field's label text
+   */
+  hideLabel?: boolean;
+  /**
+   * Id prop passed to the FieldText pebble component
+   */
+  id?: string;
+  /**
+   * Flag that will make the input read-only when true
+   */
+  isReadOnly?: boolean;
+  /**
+   * Text that is displayed as the field's label
+   */
+  label?: string;
+  /**
+   * The reference to the field name that exists in Formik state
+   */
+  name: string;
+  /**
+   * Map of the existing touched fields in the Formik form state
+   */
+  touched?: object;
+  /**
+   * The input type that follows the HTML5 standard (i.e. text, number, email, password)
+   */
+  type?: string;
+  /**
+   * Additional hint displayed beneath the input
+   */
+  helpText?: string;
+  /**
+   * Callback provided by Formik Field render props to fire when input is blurred
+   */
+  onBlur?: Function;
+  /**
+   * Callback provided by Formik Field render props to fire when input is changed
+   */
+  onChange?: Function;
+  /**
+   * Callback function called when input is focused
+   */
+  onFocus?: Function;
+  /**
+   * Text that is displayed in the field when it is empty
+   */
+  placeholder?: string;
+  /**
+   * Text or node to display before the value
+   */
+  prefix?: string | JSX.Element;
+  /**
+   * If the section is required
+   */
+  required?: boolean;
+  /**
+   * Changes the size of the input, giving it more or less padding
+   * and font size
+   */
+  size?: 'small' | 'medium' | 'large';
+  /**
+   * Text or node to display after the value
+   */
+  suffix?: string | JSX.Element;
+  /**
+   * The current field value for the input
+   */
+  value?: string;
```

Comment:
```suggestion
  value: string;
```
I making this not required would like lead to the "uncontrolled to controlled component error"